lib/EnzymeCore/src/EnzymeCore.jl

```diff
@@ -578,4 +578,88 @@ Return a new mode with its [`ABI`](@ref) set to the chosen type.
 """
 function set_abi end
 
+"""
+    split_mode(::ReverseMode)
+
+Turn a [`ReverseMode`](@ref) object into a [`ReverseModeSplit`](@ref) object while preserving as many of the settings as possible.
+
+This function acts as the identity on a [`ReverseModeSplit`](@ref).
+
+See also [`unsplit_mode`](@ref).
+"""
+function split_mode(::ReverseMode{
+    ReturnPrimal,
+    RuntimeActivity,
+    ABI,
+    Holomorphic,
+    ErrIfFuncWritten
+}) where {
+    ReturnPrimal,
+    RuntimeActivity,
+    ABI,
+    Holomorphic,
+    ErrIfFuncWritten
+}
+    ReturnShadow = true
+    Width = 0
```

Comment:
Can these be (perhaps default) args to this fn?

Reply:
I added them as positional arguments as `Val` types with default values, to ensure type stability